src/diffpy/pdfgui/control/mpdfcalculator.py

```diff
@@ -0,0 +1,892 @@
+#!/usr/bin/env python
+##############################################################################
+#
+# diffpy.mpdf       by Billinge Group
+#                     Simon J. L. Billinge sb2896@columbia.edu
+#                     (c) 2016 trustees of Columbia University in the City of
+#                           New York.
+#                      All rights reserved
+#
+# File coded by:    Benjamin Frandsen
+#
+# See AUTHORS.txt for a list of people who contributed.
+# See LICENSE.txt for license information.
+#
+##############################################################################
+
+"""functions and classes to perform mPDF calculations"""
+
+import copy
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy.signal import convolve, fftconvolve
+
+def jCalc(q, params=[0.2394, 26.038, 0.4727, 12.1375, 0.3065, 3.0939, -0.01906],
+          j2=False):
+    """Calculate the magnetic form factor j0.
+
+    This method for calculating the magnetic form factor is based on the
+    approximate empirical forms based on the tables by Brown, consisting of
+    the sum of 3 Gaussians and a constant.
+
+    Args:
+        q (numpy array): 1-d grid of momentum transfer on which the form
+            factor is to be computed
+        params (python list): provides the 7 numerical coefficients. The
+            default is an average form factor of 3d j0 approximations.
+        j2 (boolean): if True, calculate the j2 approximation for orbital
+            angular momentum contributions
+
+    Returns:
+        numpy array with same shape as q giving the magnetic form factor j0 or j2.
+    """
+    [A, a, B, b, C, c, D] = params
+    if j2:
+        return (A*np.exp(-a*(q/4/np.pi)**2)+B*np.exp(-b*(q/4/np.pi)**2)+C*np.exp(-c*(q/4/np.pi)**2)+D)*(q/4.0/np.pi)**2
+    else:
+        return A*np.exp(-a*(q/4/np.pi)**2)+B*np.exp(-b*(q/4/np.pi)**2)+C*np.exp(-c*(q/4/np.pi)**2)+D
+
+def cv(x1, y1, x2, y2):
+    """Perform the convolution of two functions and give the correct output.
+
+
+    Args:
+        x1 (numpy array): independent variable of first function; must be in
+            ascending order
+        y1 (numpy array): dependent variable of first function
+        x2 (numpy array): independent variable of second function; must have
+            same grid spacing as x1
+        y2 (numpy array): dependent variable of second function
+
+    Returns:
+        xcv (numpy array): independent variable of convoluted function, has
+            dimension len(x1) + len(x2) - 1
+
+        ycv (numpy array): convolution of y1 and y2, same shape as xcv
+
+    """
+    dx = x1[1]-x1[0]
+    ycv = dx*convolve(y1, y2, 'full')
+    xcv = np.linspace(x1[0]+x2[0], x1[-1]+x2[-1], len(ycv))
+    return xcv, ycv
+
+def fourierTransform(q, fq, rmin=0.0, rmax=50.0, rstep=0.1): # requires even q-grid
```

Comment:
why do we write the FT function from stretch instead of importing mature packages or reusing PDFgui's codes?

Meanwhile, what is the performance for this function? How long would it take to run?